microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/ls/MicroProfileDelegateCommandHandlerForJava.java

```diff
@@ -50,10 +80,67 @@ public Object executeCommand(String commandId, List<Object> arguments, IProgress
 			return getCodeLensForJava(arguments, commandId, progress);
 		case JAVA_HOVER_COMMAND_ID:
 			return getHoverForJava(arguments, commandId, progress);
+		case "projectIdentifier/identify":
+			return getProjectIdentifierResults(arguments, progress);
 		default:
 			throw new UnsupportedOperationException(String.format("Unsupported command '%s'!", commandId));
 		}
 	}
+	
+	private static List<String> getProjectPathsFromArgs(List<Object> arguments) {
+		Map<String, Object> obj = getFirst(arguments);
+		if (obj == null) {
+			throw new UnsupportedOperationException();
+		}
+		List<String> projectPaths = (List<String>) getList(obj, "projectPaths");
+		if (projectPaths == null) {
+			throw new UnsupportedOperationException();
+		}
+		return projectPaths;
+		
+	}
+
+	private static ProjectIdentifierResults getProjectIdentifierResults(List<Object> arguments, IProgressMonitor monitor) throws IOException, CoreException {		
+		List<String> projectPaths =  getProjectPathsFromArgs(arguments);
+		ProjectIdentifierResults results = new ProjectIdentifierResults();
+		
+		for (String path: projectPaths) {
+			results.addResult(path, getProjectLabel(path));
+		}
+		return results;
+	}
+	
+	private static boolean isMavenOrGradleProject(IProject project) throws CoreException {
+		return project.hasNature("org.eclipse.m2e.core.maven2Nature") || project.hasNature("org.eclipse.buildship.core.gradleprojectnature");
+	}
+	
+	private static String getProjectLabel(String path) throws CoreException {
+		IPath projectPath = new Path(path + "/.project");
+		IProjectDescription description = ResourcesPlugin.getWorkspace().loadProjectDescription(projectPath);
+		IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(description.getName());
+		
+		if (!isMavenOrGradleProject(project)) {
```

Comment:
Do we need to check that? If project is not a m2e project, is it working? If in the future Quarkus have a new build support different from maven and gradle, it will not work.

Reply:
After talking with @fbricon, I removed this in my latest push.

But at the moment, there is no code that determines whether the project is a Maven or Gradle project. Is there a robust way of doing that without using `IProject.hasNature()`?